### PR TITLE
Add index creation/drop table lock linting

### LINKED ISSUE #99: Raise backend specific deployment implications
For instance, certain operations will potentially lock a table, which can have implications during deployment (lots of operations require the table => we don't acquire the lock waiting to migrate / one we have the lock, long migration, locking the table and making production hang)

https://github.com/yandex/zero-downtime-migrations might be interesting to help out for these warnings

### PR DESCRIPTION
**Pull request by David-Wobrock**
Fixes #99

### CODE REVIEW

File: tests/unit/test_sql_analyser.py [hunk 1]
```diff
@@ -13,14 +13,22 @@ def analyse_sql(self, sql):
...
+        errors, _, warnings = self.analyse_sql(sql)
         self.assertEqual(0, len(errors), "Found errors in sql: {}".format(errors))
+        if not allow_warnings:
+            self.assertEqual(
+                0, len(warnings), "Found warnings in sql: {}".format(errors)
+            )
 
     def assertBackwardIncompatibleSql(self, sql):
-        errors, _ = self.analyse_sql(sql)
+        errors, _, _ = self.analyse_sql(sql)
```
Comment (skarzi) [inline tests/unit/test_sql_analyser.py:25]: NIP: or even shorter:

```suggestion
        errors, *_ = self.analyse_sql(sql)
```

File: tests/unit/test_sql_analyser.py [hunk 2]
```diff
@@ -13,14 +13,22 @@ def analyse_sql(self, sql):
...
         self.assertNotEqual(0, len(errors), "Found no errors in sql")
 
+    def assertWarningSql(self, sql):
+        _, _, warnings = self.analyse_sql(sql)
```
Comment (skarzi) [inline tests/unit/test_sql_analyser.py:29]:
```suggestion
       * _, warnings = self.analyse_sql(sql)
```

File: tests/unit/test_sql_analyser.py [hunk 3]
```diff
@@ -203,3 +211,23 @@ def test_make_column_not_null_with_lib_default(self):
             'ALTER TABLE "app_drop_default_a" ALTER COLUMN "col" SET DEFAULT `\'empty\';',
         ]
         self.assertValidSql(sql)
+
+    def test_create_index_non_concurrently(self):
+        sql = "CREATE INDEX ON films ((lower(titre)));"
```
Comment (skarzi) [inline tests/unit/test_sql_analyser.py:216]:
```suggestion
        sql = "CREATE INDEX ON films ((lower(title)));"
```

File: tests/unit/test_sql_analyser.py [hunk 4]
```diff
@@ -203,3 +211,23 @@ def test_make_column_not_null_with_lib_default(self):
...
+        self.assertWarningSql(sql)
+        sql = "CREATE UNIQUE INDEX title_idx ON films (title);"
+        self.assertWarningSql(sql)
+
+    def test_create_index_concurrently(self):
+        sql = "CREATE INDEX CONCURRENTLY ON films ((lower(titre)));"
```
Comment (skarzi) [inline tests/unit/test_sql_analyser.py:222]:
```suggestion
        sql = "CREATE INDEX CONCURRENTLY ON films ((lower(title)));"
```

File: tests/unit/test_sql_analyser.py (hunk 1 above)
Comment (David-Wobrock) [reply tests/unit/test_sql_analyser.py:25]: We eventually will want to refactor the return to container object that will hold all fields.
However, I do prefer this failing for now when we change the number of returned arguments

File: tests/unit/test_sql_analyser.py (hunk 2 above)
Comment (David-Wobrock) [reply tests/unit/test_sql_analyser.py:29]: Same here

File: django_migration_linter/migration_linter.py [hunk 5]
```diff
@@ -159,17 +159,19 @@ def lint_migration(self, migration):
...
         sql_statements = self.get_sql(app_label, migration_name)
-        errors, ignored = analyse_sql_statements(
+        errors, ignored, warnings = analyse_sql_statements(
             sql_statements,
             settings.DATABASES[self.database]["ENGINE"],
             self.exclude_migration_tests,
         )
 
-        err, ignored_data, warnings = self.analyse_data_migration(migration)
+        err, ignored_data, warnings_data = self.analyse_data_migration(migration)
```
[the remark this answers is not in the document]
Comment (David-Wobrock) [reply django_migration_linter/migration_linter.py:168]: We definitely will want to refactor that to an object containing all these fields :)
But we'll do it once we'll drop Python 2 and we are able to use `NamedTuple` or even `dataclass`

Maybe in some months :muscle:

**Comment by David-Wobrock** [comment]
Thanks for the review @skarzi ! :muscle: Feel free to comment more, but be aware that I'm still working on this branch.
So it might still get quite some changes ;)

**Pull request merged**